actions/databases.go

```diff
@@ -0,0 +1,290 @@
+package actions
+
+import (
+	"errors"
+	"log"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/gobuffalo/buffalo"
+)
+
+// DatabasesList gets a list of databases for a given account
+//   If the `all=true` parameter is passed it will return a list of clusters in addition to instances.
+func DatabasesList(c buffalo.Context) error {
+	// if all param is given, we'll return information about both instances and clusters
+	// otherwise, only database instances will be returned
+	all := false
+	if b, err := strconv.ParseBool(c.Param("all")); err == nil {
+		all = b
+	}
+
+	rdsClient := RDS[c.Param("account")]
+	var clustersOutput *rds.DescribeDBClustersOutput
+	var instancesOutput *rds.DescribeDBInstancesOutput
+	var err error
+
+	if all {
+		clustersOutput, err = rdsClient.Service.DescribeDBClusters(&rds.DescribeDBClustersInput{})
+		if err != nil {
+			log.Println(err.Error())
+		}
+	}
+
+	instancesOutput, err = rdsClient.Service.DescribeDBInstances(&rds.DescribeDBInstancesInput{})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			log.Println(aerr.Error())
+			return c.Error(400, aerr)
+		} else {
+			log.Println(err.Error())
+			return err
+		}
+	}
+
+	output := struct {
+		*rds.DescribeDBClustersOutput
+		*rds.DescribeDBInstancesOutput
+	}{
+		clustersOutput,
+		instancesOutput,
+	}
+	return c.Render(200, r.JSON(output))
```

Comment:
do you want your output to be separated too?  this will merge the structs i think.  also, you could just create the anonymous struct inside of the return but that may get confusing.

Reply:
yeah, it's merging my structs but it works out well, since I get the original RDS structs:
```
{
    "DBClusters": [
        { cluster1 }
    ],
    "DBInstances": [
        { instance1 }
    ]
}
```

Reply:
I wasn't sure if we only want to return the ARNs or names for the list since we're already getting all the details. Probably depends on how we're going to use that endpoint.

Reply:
yeah writing the client might inform this, although in general we give a list of id's for the list endpoint which i think makes sense